mcs/class/corlib/System.Security.Policy/Evidence.cs

```diff
@@ -260,16 +260,14 @@ static internal Evidence GetDefaultHostEvidence (Assembly a)
 
 			// Authenticode(r) signed assemblies get a Publisher evidence
 			if (IsAuthenticodePresent (a)) {
-				// Note: The certificate is part of the evidences even if it is not trusted!
-				// so we can't call X509Certificate.CreateFromSignedFile
-				AuthenticodeDeformatter ad = new AuthenticodeDeformatter (a.Location);
-				if (ad.SigningCertificate != null) {
-					X509Certificate x509 = new X509Certificate (ad.SigningCertificate.RawData);
-					if (x509.GetHashCode () != 0) {
-						e.AddHost (new Publisher (x509));
-					}
+				try {
+					X509Certificate x509 = X509Certificate.CreateFromSignedFile (a.Location);
+					e.AddHost (new Publisher (x509));
+				}
+				catch (CryptographicException) {
 				}
 			}
```

Comment:
The comment about X509Certificate.CreateFromSignedFile is totally bogus. AuthenticodeDeformatter checks the signature and the result is ignored. X509Certificate.CreateFromSignedFile is supposed to only load the certificate, but not validate its chain, so in theory it's a weaker verification.

X509Certificate.CreateFromSignedFile is currently non-working in master, in the same way as it is broken in CoreFX. That's a separate issue though.

Reply:
Let's submit that to CoreFX as we're now using their implementation.

Reply:
@baulig I have a patch queue for CoreFX. It's a bit overflowing due to all the vacations of other people and some blocking issues, but I am aware of this and I do plan to fix it.